x/sequencer/keeper/sequencer.go

```diff
@@ -61,3 +93,58 @@ func (k Keeper) GetAllSequencer(ctx sdk.Context) (list []types.Sequencer) {
 
 	return
 }
+
+// GetSequencersByRollapp returns a sequencersByRollapp from its index
+func (k Keeper) GetSequencersByRollappByStatus(ctx sdk.Context, rollappId string, status types.OperatingStatus) (list []types.Sequencer) {
+	if status == types.Unspecified {
+		return k.GetSequencersByRollapp(ctx, rollappId)
+	}
+	prefixKey := types.SequencersByRollappByStatusKey(rollappId, status)
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), prefixKey)
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+
+	defer iterator.Close() // nolint: errcheck
+
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.Sequencer
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		list = append(list, val)
+	}
+
+	return
+}
+
+/* -------------------------------------------------------------------------- */
+/*                               Unbonding queue                              */
+/* -------------------------------------------------------------------------- */
+
+// GetMatureUnbondingSequencers returns all unbonding sequencers
+func (k Keeper) GetMatureUnbondingSequencers(ctx sdk.Context, endTime time.Time) (list []types.Sequencer) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := store.Iterator(types.UnbondingQueueKey, sdk.PrefixEndBytes(types.UnbondingQueueByTimeKey(endTime)))
+
+	defer iterator.Close() // nolint: errcheck
+
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.Sequencer
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		list = append(list, val)
+	}
+
+	return
+}
+
+func (k Keeper) setUnbondingSequencerQueue(ctx sdk.Context, sequencer types.Sequencer) {
+	store := ctx.KVStore(k.storeKey)
+	b := k.cdc.MustMarshal(&sequencer)
+
+	unbondingQueueKey := types.UnbondingSequencerKey(sequencer.SequencerAddress, sequencer.UnbondTime)
+	store.Set(unbondingQueueKey, b)
+}
+
+// remove unbonding sequencer from the queue
+func (k Keeper) removeUnbondingSequencer(ctx sdk.Context, sequencer types.Sequencer) {
+	store := ctx.KVStore(k.storeKey)
+	unbondingQueueKey := types.UnbondingSequencerKey(sequencer.SequencerAddress, sequencer.UnbondTime)
```

Comment:
What about the other keys this sequencer is saved under? e.g `SequencerByRollappByStatusKey` and `SequencerKey`. shouldn't we delete them also?

Reply:
no. I don't want to delete the sequencer object
just to avoid duplicate, as it stored under `Unbonded` status
I'm deleting the `Unbonding` key prefix